### PR TITLE
feat: Add client events to did exchange bddtest basic flow

### LINKED ISSUE #394: Add client events to did exchange bddtest basic flow
depend on #199

### PR DESCRIPTION
**Pull request by fqutishat**
closes #394

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: test/bdd/features/didexchange_e2e.feature [hunk 1]
```diff
@@ -11,7 +11,11 @@ Feature: Decentralized Identifier(DID) exchange between the agents
   @didexchange_e2e
   Scenario: did exchange e2e flow
     Given "Alice" agent is running on "localhost" port "random"
+    And   "Alice" register to receive notification for post state event "completed"
     Given "Bob" agent is running on "localhost" port "random"
+    And   "Bob" register to receive notification for post state event "completed"
     And   "Alice" creates invitation
     And   "Bob" receives invitation from "Alice"
+    And   "Alice" wait for post state event "completed"
```
Comment (sandrask) [inline test/bdd/features/didexchange_e2e.feature]: "Alice" waits
"Bob" waits

Comment (fqutishat) [reply test/bdd/features/didexchange_e2e.feature]: done

File: go.sum [hunk 2]
```diff
@@ -1,4 +1,5 @@
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/DATA-DOG/godog v0.7.13 h1:JmgpKcra7Vf3yzI9vPsWyoQRx13tyKziHtXWDCUUgok=
```
Comment (troyronda) [inline go.sum]: This shouldn’t be in the root go.mod

File: test/bdd/agent_steps.go [hunk 3]
```diff
@@ -72,9 +76,36 @@ func (a *AgentSteps) createAgent(agentID, inboundHost, inboundPort string) error
...
 
+func (a *AgentSteps) registerPostMsgEvent(agentID, statesValue string) error {
+	statusCh := make(chan dispatcher.StateMsg, 10)
+	if err := a.bddContext.DIDExchangeClients[agentID].RegisterMsgEvent(statusCh); err != nil {
+		return fmt.Errorf("failed to register msg event: %w", err)
+	}
+	states := strings.Split(statesValue, ",")
+	for _, state := range states {
+		a.bddContext.PostStatesFlag[state] = make(chan bool)
+	}
```
Comment (troyronda) [inline test/bdd/agent_steps.go]: Might be an idea to extract a function.

File: go.sum (hunk 2 above)
Comment (fqutishat) [reply go.sum]: done

File: test/bdd/agent_steps.go (hunk 3 above)
Comment (fqutishat) [reply test/bdd/agent_steps.go]: done

**Pull request merged**